### DISCUSSION TITLE
Getting "There is X other session using the database." with Supabase Supavisor

### DISCUSSION THREAD

**Discussion by danielcopa** [prisma/prisma#22793] [Q&A] [answered]
### Question

Getting an error when trying a migration againts Supabase Supavisor: 

```
Error: db error: ERROR: database "prisma_migrate_shadow_db_399..." is being accessed by other users
DETAIL: There is 1 other session using the database.
   0: schema_core::state::DevDiagnostic
             at schema-engine/core/src/state.rs:270
```

### How to reproduce (optional)

1. Copy your "Session" pooler database url
2. Set it as directUrl on your schema.prisma
3. Try to run a migration

### Information about Prisma Schema, Client Queries and Environment (optional)

prisma                  : 5.8.1
@prisma/client          : 5.8.1
Computed binaryTarget   : darwin-arm64
Operating System        : darwin
Architecture            : arm64
Node.js                 : v20.5.1
Query Engine (Node-API) : libquery-engine 78caf6feeaed953168c64e15a249c3e9a033ebe2 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli 78caf6feeaed953168c64e15a249c3e9a033ebe2 (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.8.1-1.78caf6feeaed953168c64e15a249c3e9a033ebe2
Default Engines Hash    : 78caf6feeaed953168c64e15a249c3e9a033ebe2
Studio                  : 0.497.0
Preview Features        : fullTextIndex, fullTextSearch

**Comment by ludralph** [comment]
Hi @danielcopa 👋 

Thank you for raising this question.

For migrations you should optimally use the direct connection. The connection pooler can keep connections to the temporary shadow database open and cause error message like the one you have reported.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by heyBett** [reply, marked as answer]
It was a bug and was corrected on the last few hours.
https://github.com/orgs/supabase/discussions/17817#discussioncomment-8228636

Thank you for the attention anyway @ludralph!
